stack.py

```diff
@@ -0,0 +1,29 @@
+class StackItem(object):
+    def __init__(self, data, next_item=None):
+        self.data = data
+        self.next_item = next_item
+
+    def __str__(self):
+        return self.data
+
+
+class StackFrame(object):
+    def __init__(self, first_item=None):
+        self.first_item = first_item
+
+    def push(self, new_item):
+        # push new_item to beginning of list
+        if not self.first_item:
+            self.first_item = new_item
+        else:
+            new_item.next_item = self.first_item
+            self.first_item = new_item
```

Comment:
If you push values, rather than items, this can be one line, like it is in the linked list.